packages/playwright-ct-react/registerSource.mjs

```diff
@@ -77,17 +79,39 @@ window.playwrightMount = async (component, rootElement, hooksConfig) => {
       App = () => wrapper;
   }
 
-  ReactDOM.render(App(), rootElement);
+  if (rootRegistry.has(rootElement)) {
+    throw new Error(
+        'Attempting to mount a component into an container that already has a React root. ' +
+        'Either unmount the React root first with `playwrightUnmount(rootElement)` or update the root instead with `playwrightUpdate(rootElement, component)`'
+    );
+  }
+
+  const root = createRoot(rootElement);
+  rootRegistry.set(rootElement, root);
+  // TODO: Do we need to flush sync?
+  root.render(App());
 
   for (const hook of /** @type {any} */(window).__pw_hooks_after_mount || [])
     await hook({ hooksConfig });
```

Comment:
I don't know what the expected semantics for the hooks are. Without `flushSync`, the hooks effectively run before React commits anything to the screen. However, it is quite nice to not have to use [`flushSync`](https://beta.reactjs.org/reference/react-dom/flushSync) since usage of `flushSync` is uncommon. It would be preferable to test the common usage of components.